### ISSUE TITLE
Improve contributing docs

### ISSUE THREAD

**Issue by shubheksha** [heptio/ark#867]
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
The current docs for contributing to the project are missing a lot of relevant information. We should add stuff around:
- installing deps
- installing/building ark locally
- testing
- various make targets we have

Some of this information is present in: https://heptio.github.io/ark/v0.9.0/quickstart, but I think it makes sense to consolidate it in `Contributing.md`.

**Comment by shubheksha** [comment]
Happy to send a PR for this if this is something y'all are open to :)
/cc @ncdc @skriss @carlisia

**Comment by ncdc** [comment]
Yes please, and thanks! We already have docs on building & running ark locally, but maybe they're not easy to find?

**Comment by nrb** [comment]
@shubheksha Definitely welcome your improvements, especially since you went through the new contributor process so recently!

**Comment by carlisia** [comment]
I love this idea. I personally would love to see a (much more concise) sort of version of this: https://github.com/kubernetes/community/tree/master/contributors/guide

I especially like the `Before you get started` and the Contributing` top sessions. We don't have a style guide or code convention, so much of that can be skipped. Of particular interest though would be a subsection on testing, running make targets (mainly the things you listed), issue management (an opportunity to highlight our use of zenhub).

We should also add a `Community` section that could point to our new https://github.com/heptio/ark-community repo, as well as all the community spaces listed here: https://github.com/heptio/ark-community/blob/master/meetings/2018/design-session/0927/README.md#heptio-ark-community (minus the zenhub).

**Comment by carlisia** [comment]
@heptio/ark-team what do you think 👆

**Comment by Bradamant3** [comment]
@shubheksha this is something we've been missing forever. Right now the information about installing locally is not really sufficient -- it's aimed at users, not contributors -- and the information about building is sort of half-and-half. I have some ideas about conceptual information to include in a contributing/developer guide, but if you'd like to take on the content for this issue, that would be awesome. Let me know how I can help!
